core/communication_manager.py

```diff
@@ -95,10 +95,11 @@ def _create_session(self, payload):
         # only considering the 'FederatedAveragingOptimizer' for now.
         # TODO: We need to incorporate session id's when we're ready.
         if not self.dataset_manager:
-            raise Exception("Dataset Manager has not been set. Communication \
-             Manager needs to be configured first!")
-        logging.info("New optimizer session is being set up...")
+            raise Exception("Dataset Manager has not been set. Communication Manager needs to be configured first!")
+        assert payload.get(TxEnum.KEY.name) is MessageEventTypes.NEW_SESSION.name, \
+            "Expected a new session but got {}".format(payload.get(TxEnum.KEY.name))
         initialization_payload = payload.get(TxEnum.CONTENT.name)
+        # logging.info("New optimizer session is being set up...{}".format(initialization_payload))
```

Comment:
Did you intend to leave this commented out?

Reply:
Nice catch